Fusion environment variables (#8461)
fishtown-analytics/docs.getdbt.com: website/docs/docs/configure-dbt-extension.md

</TabItem>
</Tabs>

#### Configure in the VS Code extension settings

To use the dbt extension menu actions/buttons, you can configure environment variables directly in the [VS Code User Settings](vscode://settings/dbt.environmentVariables) interface or in a `.env` file in your current working directory. This includes both your custom variables and any automatic [<Constant name="dbt_platform"/> variables](/docs/build/environment-variables) (like `DBT_CLOUD_ENVIRONMENT_NAME`) that your project depends on.

- Configure variables in the VS Code **User Settings** or in a `.env` file to have them recognized by the extension. For example, when using <Term id="lsp" /> -powered features, "Show build menu," and more.
- VS Code does not inherit variables set by the VS Code terminal or external shells.
- The terminal uses system environmental variables, and does not inherit variables set in the dbt VS Code extension config. For example, running a dbt command in the terminal won't fetch or use the dbt VS Code extension variables.

> :::info `.env` file support

Reviewer: This is a lot of information for a callout. This might better serve readers if we make it a section instead - maybe an additional H4?

Author: good shout - i had it as an h4 and then thought it was too much and reverted back to a callout . i'll update it

> Both the [<Constant name="fusion"/> CLI](/docs/fusion/install-fusion-cli) and the dbt VS Code extension will automatically read environment variables from a `.env` file in your current working directory (the folder you `cd` into and run commands from in your terminal), if one exists. This means environment variables you define in `.env` will be available when running dbt commands in the terminal as well as when using the extension's menu actions. 

Reviewer: ```suggestion
The [<Constant name="fusion"/> CLI](/docs/fusion/install-fusion-cli) and the dbt VS Code extension can automatically read environment variables from a `.env` file in your current working directory (the folder you `cd` into and run dbt commands from in your terminal), if one exists. The environment variables you define in the `.env` file are available both when running dbt commands in the terminal as well as when using the extension's menu actions. 
```

> Here's some more information about the `.env` file:

Reviewer: Since this is restating the exact title of the expandable header we can probably remove
```suggestion
```

<Expandable alt_header="More information about the .env file">

- The `.env` file provides a convenient way to set environment variables that work across both the CLI and the extension.

> - The `.env` file is loaded _only_ from your current working directory. It doesn't support the `--project-dir` flag or `DBT_PROJECT_DIR` environment variable and doesn't walk up to find your dbt project root. For best results, keep your `.env` file in your project root and run commands from there.

Reviewer: "Walk up" might be confusing to some. Is there another way we can state this point? Maybe search?
```suggestion
- We recommend placing your `.env` file in the project root and running dbt commands from that location because the file is loaded _only_ from your current working directory. It doesn't support the `--project-dir` flag or `DBT_PROJECT_DIR` environment variable, and dbt won't search your project root if you're running commands from a different directory location.

> - Add `.env` to your `.gitignore` file to prevent sensitive credentials from being committed to version control.

Reviewer: ```suggestion
- Add `.env` to your `.gitignore` file to prevent sensitive credentials from being committed to your repo.
```
